sys-libs/libcxx/libcxx-9999.ebuild

```diff
@@ -122,49 +76,42 @@ END_LDSCRIPT
 }
 
 gen_static_ldscript() {
-	if use libcxxrt ; then
-		# Move it first.
-		mv "${ED}/usr/$(get_libdir)/libc++.a" "${ED}/usr/$(get_libdir)/libc++_static.a" || die
-
-		# Generate libc++.a ldscript for inclusion of its dependencies so that
-		# clang++ -stdlib=libc++ -static works out of the box.
-		local deps="${EPREFIX}/usr/$(get_libdir)/libc++_static.a ${EPREFIX}/usr/$(get_libdir)/libcxxrt.a"
-		# On Linux/glibc it does not link without libpthread or libdl. It is
-		# fine on FreeBSD.
-		use elibc_glibc && deps="${deps} ${EPREFIX}/usr/$(get_libdir)/libpthread.a ${EPREFIX}/usr/$(get_libdir)/libdl.a"
-
-		# unlike libgcc_s, libunwind is not implicitly linked
-		use libunwind && deps="${deps} ${EPREFIX}/usr/$(get_libdir)/libunwind.a"
-
-		gen_ldscript "${deps}" > "${ED}/usr/$(get_libdir)/libc++.a"
-	fi
-	# TODO: Generate a libc++.a ldscript when building against libsupc++
+	local libdir=$(get_libdir)
+	local cxxabi=libcxxrt
+
+	# Move it first.
+	mv "${ED}/usr/${libdir}/libc++.a" "${ED}/usr/${libdir}/libc++_static.a" || die
+
+	# Generate libc++.a ldscript for inclusion of its dependencies so that
+	# clang++ -stdlib=libc++ -static works out of the box.
+	local deps="${EPREFIX}/usr/${libdir}/libc++_static.a ${EPREFIX}/usr/${libdir}/${cxxabi}.a"
+	# On Linux/glibc it does not link without libpthread or libdl. It is
+	# fine on FreeBSD.
+	use elibc_glibc && deps="${deps} ${EPREFIX}/usr/${libdir}/libpthread.a ${EPREFIX}/usr/${libdir}/libdl.a"
+	# unlike libgcc_s, libunwind is not implicitly linked
+	use libunwind && deps="${deps} ${EPREFIX}/usr/${libdir}/libunwind.a"
+
+	gen_ldscript "${deps}" > "${ED}/usr/${libdir}/libc++.a"
 }
 
 gen_shared_ldscript() {
-	if use libcxxrt ; then
-		mv "${ED}/usr/$(get_libdir)/libc++.so" "${ED}/usr/$(get_libdir)/libc++_shared.so" || die
-		local deps="${EPREFIX}/usr/$(get_libdir)/libc++_shared.so ${EPREFIX}/usr/$(get_libdir)/libcxxrt.so"
-		use libunwind && deps="${deps} ${EPREFIX}/usr/$(get_libdir)/libunwind.so"
-		gen_ldscript "${deps}" > "${ED}/usr/$(get_libdir)/libc++.so"
-	fi
-	# TODO: Generate the linker script for other configurations too.
+	local libdir=$(get_libdir)
+	local cxxabi=libcxxrt
+
+	mv "${ED}/usr/${libdir}/libc++.so" "${ED}/usr/${libdir}/libc++_shared.so" || die
+	local deps="${EPREFIX}/usr/${libdir}/libc++_shared.so ${EPREFIX}/usr/${libdir}/${cxxabi}.so"
+	use libunwind && deps="${deps} ${EPREFIX}/usr/${libdir}/libunwind.so"
+	gen_ldscript "${deps}" > "${ED}/usr/${libdir}/libc++.so"
 }
 
 multilib_src_install() {
-	cd "${BUILD_DIR}/lib"
-	if use static-libs ; then
-		dolib.a libc++.a
-		gen_static_ldscript
-	fi
-	dolib.so libc++.so*
+	cmake-utils_src_install
 	gen_shared_ldscript
```

Comment:
Upstream does have mechanism for this, but it has a few problems:
- like test cases, it doesn't work in a standalone build 
- it only supports generating script for shared lib, not for static lib
- it doesn't take libunwind into consideration

Some heavy patching might help, but I didn't try.